### PR TITLE
foot: 1.4.4 -> 1.5.1

### PR DESCRIPTION
**Pull request by SeTSeR**
###### Motivation for this change

Update foot to latest version. It also required to update some of foot's dependencies and add new dependency for fcft.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/foot/default.nix [hunk 1]
```diff
@@ -5,12 +5,12 @@
 
 stdenv.mkDerivation rec {
   pname = "foot";
-  version = "1.4.4";
+  version = "1.5.1";
 
   src = fetchgit {
     url = "https://codeberg.org/dnkl/foot.git";
     rev = "${version}";
```
Comment (sternenseemann) [inline pkgs/applications/misc/foot/default.nix:12]: Can be simplified:

```suggestion
    rev = version;
```

File: pkgs/development/libraries/tllist/default.nix [hunk 2]
```diff
@@ -2,12 +2,12 @@
...
   pname = "tllist";
-  version = "1.0.2";
+  version = "1.0.4";
 
   src = fetchgit {
     url = "https://codeberg.org/dnkl/tllist.git";
     rev = "${version}";
-    sha256 = "095wly66z9n2r6h318rackgl4g1w9l1vj96367ngcw7rpva9yppl";
+    sha256 = "sha256-+u8p/VmI61SGRhZHaJBwgcVNetNOrYzg2NVQehbfRqg=";
   };
```
Comment (sternenseemann) [inline pkgs/development/libraries/tllist/default.nix:12]: Also enable tests for `tllist`:

```suggestion
    doCheck = true;
    
```

File: pkgs/development/libraries/tllist/default.nix [hunk 3]
```diff
@@ -2,12 +2,12 @@
 
 stdenv.mkDerivation rec {
   pname = "tllist";
-  version = "1.0.2";
+  version = "1.0.4";
 
   src = fetchgit {
     url = "https://codeberg.org/dnkl/tllist.git";
     rev = "${version}";
```
Comment (sternenseemann) [inline pkgs/development/libraries/tllist/default.nix:9]:
```suggestion
    rev = version;
```

File: pkgs/development/libraries/fcft/default.nix [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
+,freetype, fontconfig, harfbuzz, pixman, tllist, check }:
 
 stdenv.mkDerivation rec {
   pname = "fcft";
-  version = "2.2.6";
+  version = "2.3.1";
 
   src = fetchgit {
     url = "https://codeberg.org/dnkl/fcft.git";
     rev = "${version}";
```
Comment (sternenseemann) [inline pkgs/development/libraries/fcft/default.nix:10]:
```suggestion
    rev = version;
```

**Review by sternenseemann — CHANGES_REQUESTED** [review]
Totally missed foot 1.5.1 and updating #98270.

Commented some smaller clean ups you could do in this changes. Enabling tests for tllist would be good as well.

Also squash the two fcft commits into one please.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 100345 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>4 packages built:</summary>  <ul>    <li>fcft</li>    <li>foot</li>    <li>fuzzel</li>    <li>tllist</li>  </ul></details>

**Pull request merged**